src/wovnio/wovnphp/Headers.php

```diff
@@ -346,4 +346,18 @@ public function shouldRedirect()
 
         return $cookieLang && ($requestLang !== $cookieLang) && ($requestLang === $this->store->defaultLang());
     }
+
+    public function getCanonicalUrl()
+    {
+        $canonicalUrl = $this->protocol . '://' . $this->host . $this->pathnameKeepTrailingSlash;
+        if ($this->store->settings['query_canonical_significance'] && !empty($this->query)) {
+            $canonicalUrl = $canonicalUrl . '?' . $this->query;
+        }
+        return $canonicalUrl;
+    }
+
+    public function isSearchEngineBot()
+    {
+        return strpos($this->env['HTTP_USER_AGENT'], 'Googlebot/2.1') !== false;
```

Comment:
checking for `/2.1` seems too specific. We don't want to update this every googlebot release.

Reply:
I thought about this but there are other bots that we do not want to offer special treatment such as `Googlebot-Image/1.0` and `Googlebot-Video/1.0`...

Reply:
I guess we can match `Googlebot/` tho. Should be fine.